File: task-for-one-neuron.fixed.go
package main
import "fmt"


var problem = [] string {
    "...?.....?............#",
    "......?.......?....####",
    ".?........?.....#####?#",
    ".......?.....#####?####",
    "..?..?....#############",
    ".......######?######?##",
    "....#####?#######?#####",
    ".###################?##" }

var wx = 8
var wy = 24
var wb = -171

func neuron(x, y int) int {
    var bias=1
    var f = wx * x + wy * y + wb * bias
    if f < 0 {
        return -1
    }
    return 1
}

func main() {
	for y, line := range problem {
		var new_line string
		for x, ch := range line {
			var char_ = ch  // char_ := ch
			if ch == '?' {
				var r = neuron(x, y)
				if r < 0 {
					char_ = '!'
				} else {
					char_ = '%'
				}
			}
			// maybe it will be better with use  var bb bytes.Buffer , or  var sb strings.Builder
			new_line += string(char_)
		}
		fmt.Println(new_line)
	}
}

File: task-for-one-neuron.fixed.java
public class Main {
	static String[] problem = {
		"...?.....?............#",
		"......?.......?....####",
		".?........?.....#####?#",
		".......?.....#####?####",
		"..?..?....#############",
		".......######?######?##",
		"....#####?#######?#####",
		".###################?##" };
	
	static int wx = 8;
	static int wy = 24;
	static int wb = -171;
	
	static int neuron(int x, int y) {
	    int bias=1;
		int f = wx * x + wy * y + wb * bias;
		if (f < 0)
			return -1;
		return 1;
	}
	
	public static void main(String[] args) {
		int y = 0;
		for (String line : problem) {
			int x = 0;
			String new_line = "";
			for (char ch : line.toCharArray()) {  // for (int x=0; i<line.length(); ++x) { char ch = line.charAt(x);
				if (ch == '?') {
					// спрашиваем у нейрона, что он думает про эту точку
					int r = neuron(x, y);
					if (r < 0)
						ch = '!';
					else
						ch = '%';
				}
				new_line += ch;
				++x;
			}
			System.out.println(new_line);
			++y;
		}
	}
}